code/controllers/subsystem/packetnets.dm

```diff
@@ -0,0 +1,335 @@
+SUBSYSTEM_DEF(packets)
+	name = "Packets"
+	wait = 1
+	priority = FIRE_PRIORITY_PACKETS
+	flags = SS_NO_INIT|SS_KEEP_TIMING
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+
+	var/list/saymodes = list()
+	var/list/datum/radio_frequency/frequencies = list()
+
+	///All the physical networks to process
+	var/list/datum/powernet/queued_networks = list()
+	///Radio packets to process
+	var/list/queued_radio_packets = list()
+	///Tablet messages to process
+	var/list/queued_tablet_messages = list()
+	///Subspace/vocal packets to process
+	var/list/queued_subspace_vocals = list()
+
+	///The current processing lists
+	var/list/current_networks = list()
+	var/list/current_radio_packets = list()
+	var/list/current_tablet_messages = list()
+	var/list/current_subspace_vocals = list()
+
+	///Tick usage
+	var/cached_cost = 0
+	var/cost_networks = 0
+	var/cost_radios = 0
+	var/cost_tablets = 0
+	var/cost_subspace_vocals = 0
+
+	///What processing stage we're at
+	var/stage = SSPACKETS_POWERNETS
+
+/datum/controller/subsystem/packets/PreInit(timeofday)
+	for(var/_SM in subtypesof(/datum/saymode))
+		var/datum/saymode/SM = new _SM()
+		saymodes[SM.key] = SM
+	return ..()
+
+/datum/controller/subsystem/packets/stat_entry(msg)
+	msg += "RP: [length(queued_radio_packets)]|"
+	msg += "TM: [length(queued_tablet_messages)]|"
+	msg += "SSV: [length(queued_subspace_vocals)]|"
+	msg += "C:{"
+	msg += "CN:[round(cost_networks, 1)]|"
+	msg += "CR:[round(cost_radios, 1)]|"
+	msg += "CSSV:[round(cost_subspace_vocals, 1)]|"
+	msg += "}"
+
+	return ..()
+
+/datum/controller/subsystem/packets/fire(resumed)
+	if(!resumed)
+		current_networks = queued_networks.Copy()
+		current_radio_packets = queued_radio_packets.Copy()
+		current_tablet_messages = queued_tablet_messages.Copy()
+		current_subspace_vocals = queued_subspace_vocals.Copy()
+
+	var/timer = TICK_USAGE_REAL
+
+	///Network packets
+	if(stage == SSPACKETS_POWERNETS)
+		timer = TICK_USAGE_REAL
+		if(!resumed)
+			cached_cost = 0
+
+		var/datum/powernet/net
+		var/obj/machinery/power/poster
+		while(length(current_networks))
+			net = current_networks[length(current_networks)]
+			if(QDELETED(net))
+				current_networks.len--
+				queued_networks -= net
+				continue
+
+			///Cycle the packet queue
+			if(!length(net.current_packet_queue))
+				net.current_packet_queue = net.next_packet_queue.Copy()
+				net.next_packet_queue.Cut()
+
+			///No packets no problem
+			if(!length(net.current_packet_queue))
+				current_networks.len--
+				continue
+
+			for(var/datum/signal/signal as anything in net.current_packet_queue)
+				///Find the poster so we don't send the signal to it's author
+				poster = signal.author.resolve()
```

Comment:
The poster isn't in the data_nodes list. It's data terminal is. This will need to be transformed into the terminal somehow. There is no consistent way to do this. Could add a new var to signals of `last_hop` or something.

Reply:
Overwrite the author if you don't need it for any other reason.